source/Nevermore/Util/TinyTypeExtensions.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Linq;
+using System.Reflection;
+using Nevermore.Contracts;
+
+namespace Nevermore.Util
+{
+    internal static class TinyTypeExtensions
+    {
+        public static Type GetTinyTypeInterface(this Type type)
+        {
+            return type
+                .GetInterfaces()
+                .SingleOrDefault(i => i.IsGenericType && i.GetGenericTypeDefinition() == typeof(ITinyType<>));
+        }
+
+        public static object GetTinyTypeValue(this Type tinyType, object value)
+        {
+            if (value == null) 
+                throw new ArgumentNullException(nameof(value));
+            
+            var property = tinyType.GetProperty(nameof(ITinyType<object>.Value), BindingFlags.Instance | BindingFlags.Public);
```

Comment:
Slightly annoying that we need to reflect on this type every time (unless we have also have a non-generic `ITinyType` interface)

Reply:
Ah, interesting. This annoyance is required to support `int` (or other value types), right? If we were only supporting reference types, we'd get away with `value is ITinyType<object>`:

```
void Main()
{
	ContributeParameter(new TypedString("foo")); // WORKS: Unboxed value without reflection: foo
	ContributeParameter(new TypedInt(42));       // :-(    Unboxed value without reflection: UserQuery+TypedInt
	ContributeParameter("just a string");        // WORKS: Unboxed value without reflection: just a string
	ContributeParameter(100);                    // WORKS: Unboxed value without reflection: 100
}

void ContributeParameter(object value)
{
	var unboxedValue = value is ITinyType<object> tinyType ? tinyType.Value : value;

	$"Unboxed value without reflection: {unboxedValue}".Dump();
}

interface ITinyType<out T>
{
	T Value { get; }
}

class TypedString : ITinyType<string>
{
	public TypedString(string value) { Value = value; }
	public string Value { get; private set; }
}

class TypedInt : ITinyType<int>
{
	public TypedInt(int value) { Value = value; }
	public int Value { get; private set; }
}
```

Reply:
>  If we were only supporting reference types, we'd get away with value is ITinyType<object>

Correct...your thoughts on this?

Reply:
It's mainly been strings (particularly IDs) that have caught my eye to this point, but supporting value types would be neat. I think the reflection is ok 👍